order/Knaster_Tarski.lean

```diff
@@ -0,0 +1,100 @@
+import order.complete_lattice order.order_iso order.fixed_points
+
+open lattice
+
+universe u
+
+variables {α : Type u} [complete_lattice α]
+variables (f : α → α) (M : monotone f)
+
+def fixed_points : set α := { x | f x = x }
+
+namespace fixed_points
+
+def previous (x : α) : α :=
+gfp (λ z, x ⊓ f z)
+
+variable {f}
+
+theorem previous.le {x : α} : previous f x ≤ x :=
+gfp_le $ λ z hz, le_trans hz inf_le_left
+
+theorem previous.le_apply {x : α} : previous f x ≤ f (previous f x) :=
+gfp_le $ λ z hz, le_trans (le_trans hz inf_le_right) $ M $ le_gfp hz
+
+theorem previous.fixed {x : α} (H : f x ≤ x) : f (previous f x) = previous f x :=
+le_antisymm
+  (le_gfp $ le_inf (le_trans (M previous.le) H) (M $ previous.le_apply M))
+  (previous.le_apply M)
+
+variable f
+
+def next (x : α) : α :=
+lfp (λ z, x ⊔ f z)
+
+variable {f}
+
+theorem next.le {x : α} : x ≤ next f x :=
+le_lfp $ λ z hz, le_trans le_sup_left hz
+
+theorem next.apply_le {x : α} : f (next f x) ≤ next f x :=
+le_lfp $ λ z hz, le_trans (le_trans (M $ show next f x ≤ z, from lfp_le hz) le_sup_right) hz
+
+theorem next.fixed {x : α} (H : x ≤ f x) : f (next f x) = next f x :=
+le_antisymm
+  (next.apply_le M)
+  (lfp_le $ sup_le (le_trans H (M next.le)) (M $ next.apply_le M))
+
```

Comment:
what about `def next.fixed_point {x : : α} (H : x ≤ f x) : fixed_point` (in similar for previous) then you can shorten some proofs below.